src/main/java/com/adobe/aem/compgenerator/utils/DialogUtils.java

```diff
@@ -163,25 +163,34 @@ private static Element createPropertyNode(Document document, Node currentNode, P
         // Some of the properties are optional based on the different types available.
         addBasicProperties(propertyNode, property);
 
-        if (StringUtils.isNotEmpty(property.getField())
-                && !property.getType().equalsIgnoreCase("image")
-                && !property.getType().equalsIgnoreCase("multifield")
-                && !property.getType().equalsIgnoreCase(Constants.TYPE_HEADING)) {
-            propertyNode.setAttribute(Constants.PROPERTY_NAME, "./" + property.getField());
-            propertyNode.setAttribute(Constants.PROPERTY_CQ_MSM_LOCKABLE, "./" + property.getField());
+        getPrimaryFieldName(property).ifPresent(name -> {
+            propertyNode.setAttribute(Constants.PROPERTY_NAME, name);
+            propertyNode.setAttribute(Constants.PROPERTY_CQ_MSM_LOCKABLE, name);
+        });
+
+        if ("image".equalsIgnoreCase(property.getType())) {
+            Element hiddenImageNode = document.createElement(property.getField() + "ResType");
```

Comment:
Should we consolidate these lines of code, since it seems they are copy/pasted into the multifield scenario?

Reply:
consolidated.